src/adapters/scene_category.js

```diff
@@ -1,21 +1,68 @@
 import { Marker } from 'mapbox-gl--ENV';
 import constants from '../../config/constants.yml';
-import { createIcon } from '../adapters/icon_manager';
 import Telemetry from 'src/libs/telemetry';
 import { toUrl } from 'src/libs/pois';
 import { fire, listen } from 'src/libs/customEvents';
+import { poisToGeoJSON, emptyFeatureCollection } from 'src/libs/geojson';
+import { filteredPoisStyle } from 'src/adapters/pois_styles';
+import { createMapGLIcon, createIcon } from 'src/adapters/icon_manager';
 
-function getMarkerId(poi) {
-  return `marker_${poi.id}`;
-}
+const DYNAMIC_POIS_LAYER = 'poi-filtered';
 
 export default class SceneCategory {
   constructor(map) {
     this.map = map;
-    this.markers = [];
+    this.hoveredPoi = null;
+    this.hoveredMarker = new Marker({
+      element: createIcon({ disablePointerEvents: true }),
+      anchor: 'bottom',
+    });
+    this.selectedMarker = new Marker({
+      element: createIcon({}),
+      anchor: 'bottom',
+    });
+
+    createMapGLIcon('./statics/images/map/pin_map_dot.svg', 50, 60)
```

Comment:
There is a lot of code in this constructor. I think we can hide the complexity of each instruction by defining standalone functions. e.g: 
```js
this.map.on('click', handlePoiClicked).
```